src/Workspaces/Core/Portable/Workspace/Workspace.TextTracker.cs

```diff
@@ -43,10 +56,51 @@ public void Disconnect()
                 => this.TextContainer.TextChanged -= _weakOnTextChanged;
 
             private void OnTextChanged(object sender, TextChangeEventArgs e)
+            {
+                // Check if we require a delay between processing text changes to this document.
+                // If we do not require a delay OR the prior text change happened before this required delay,
+                // then we process this text change immediately.
+                // Otherwise, we kick off a Task.Delay to enforce this delay.
+                // After the delay, we check if there was another text change after the current one.
+                // If so, we don't process the current text change and wait for the next one to be processed.
+                // This ensures that only the last text change is processed, optimizing typing performance.
+
+                if (_delayBetweenProcessingTextChanges == TimeSpan.Zero)
+                {
+                    OnTextChangedCore(e.NewText);
+                    return;
+                }
+
+                var lastTextChangeTickCount = _lastTextChangeTickCount;
+                var currentTickCount = DateTime.UtcNow.Ticks;
+                _lastTextChangeTickCount = currentTickCount;
+
+                if (lastTextChangeTickCount == 0 ||
+                    currentTickCount - lastTextChangeTickCount >= _delayBetweenProcessingTextChanges.Ticks)
+                {
+                    OnTextChangedCore(e.NewText);
+                    return;
+                }
+
+                Task.Run(async delegate
+                {
+                    await Task.Delay(_delayBetweenProcessingTextChanges).ConfigureAwait(false);
+
+                    if (_lastTextChangeTickCount != currentTickCount)
+                    {
+                        // There was another subsequent text change.
+                        return;
+                    }
+
+                    OnTextChangedCore(e.NewText);
```

Comment:
Does this work also need to be cancelled if the Disconnect() method of this class is called?